### PR TITLE
Overviews rtorre touches

### PR DESCRIPTION
**Pull request by rafatower**
@jgoizueta I've written this code while taking a closer look to yours. Feel free to comment and/or merge into `overviews` branch.

Cheers!

### CODE REVIEW

**Comment by rafatower** [comment]
Some comments:
- it seems to be asymptotic both in exec times and returned results (which kind of makes sense)
- for low zoom levels it tends to underestimate the max zoom returned. From the [docs](http://www.postgis.org/docs/ST_Estimated_Extent.html):

> resulting extent will be about 95% of the real one

File: scripts-available/CDB_Overviews.sql [hunk 1]
```diff
@@ -1,3 +1,46 @@
...
+      table_id.schema_name, table_id.table_name, 'the_geom_webmercator'
+    );
+
+    BEGIN
+      EXECUTE ext_query INTO ext;
+      EXCEPTION
+        -- This is the typical ERROR: stats for "mytable" do not exist
+        WHEN internal_error THEN
+          -- Get stats and execute again
+          EXECUTE format('ANALYZE %1$I', reloid);
```
Comment (pnorman) [inline scripts-available/CDB_Overviews.sql]: `ANALYZE %1$I (the_geom_webmercator)` might be better here, as it only analyzes the column needed for this function. On the other hand, if one column is missing statistics, they probably all are, and analyzing everything could be useful.

**Comment by pnorman** [comment]
> for low zoom levels it tends to underestimate the max zoom returned

The 2-d postgis case is a bit more complicated than sampling numbers, but for large tables, postgres basically uses sampling of random pages, which means that it's likely to miss some maxima.

**Comment by rafatower** [comment]
merged as per our conversation

**Pull request merged**